app.vup.Vup.yaml

```diff
@@ -0,0 +1,78 @@
+app-id: app.vup.Vup
+runtime: org.gnome.Platform
+runtime-version: '43'
+sdk: org.gnome.Sdk
+add-extensions:
+  org.freedesktop.Platform.ffmpeg-full:
+    directory : "lib/ffmpeg"
+    version: "21.08"
+    add-ld-path: .  
+cleanup-commands:
+- mkdir -p ${FLATPAK_DEST}/lib/ffmpeg      
+command: vup
+separate-locales: false
+finish-args:
+- "--share=ipc"
+- "--socket=fallback-x11"
+- "--socket=wayland"
+- "--device=dri"
+- "--socket=pulseaudio"
+- "--share=network"
+- "--filesystem=home"
+- "--talk-name=org.kde.StatusNotifierWatcher"
+- "--talk-name=org.freedesktop.Notifications"
+- "--filesystem=xdg-run/app/com.discordapp.Discord:create"
+modules:
+- name: libjsoncpp
+  buildsystem: cmake
+  config_opts:
+  - -DCMAKE_BUILD_TYPE=release
+  - -DBUILD_SHARED_LIBS=ON
+  sources:
+  - type: archive
+    url: https://github.com/open-source-parsers/jsoncpp/archive/refs/tags/1.7.5.tar.gz
+    sha256: 4338c6cab8af8dee6cdfd54e6218bd0533785f552c6162bb083f8dd28bf8fbbe
+- name: libsodium
+  buildsystem: autotools
+  sources:
+  - type: archive
+    url: https://github.com/jedisct1/libsodium/archive/refs/tags/1.0.18-RELEASE.tar.gz
+    sha256: b7292dd1da67a049c8e78415cd498ec138d194cfdb302e716b08d26b80fecc10
+- name: intltool
+  buildsystem: autotools
+  sources: 
+  - type: archive
+    url: https://gitlab.gnome.org/World/intltool/-/archive/release-0_51_0/intltool-release-0_51_0.tar.gz
+    sha256: 3d16e312265cf92658668d1aace692e3bfa8ff08b3449a124571b320ae3a8eac
+- shared-modules/libappindicator/libappindicator-gtk3-introspection-12.10.json
+- shared-modules/libsecret/libsecret.json
+- name: vup
+  buildsystem: simple
+  build-commands:
+  - mkdir -p Vup
+  - unzip Vup-0.12.0-BETA-Linux.zip -d build
```

Comment:
I'll assume this is a x86_64 specific binary. But there is still an aarch64 to be done.

Reply:
why not building from source?

Reply:
> why not building from source?

Would it be preferred to download the entirety of flutter to then build this project? It seems much simpler to just build from a release file.

Reply:
from a release for a single architecture.

But then you mentionned Flutter...

Reply:
the problem here is that the aarch64 flatpak is built with the x86_64 binaries.

Reply:
So my only real option here would be to have a precompiled arrch64 image on the github releases page. Dart does not support cross compilation. Other flutter apps seemed to not have included aarch64 builds at all: [pstube](https://github.com/flathub/io.github.prateekmedia.pstube/blob/master/io.github.prateekmedia.pstube.yml), [spottube](https://github.com/flathub/com.github.KRTirtho.Spotube/blob/master/com.github.KRTirtho.Spotube.yml).

Reply:
this is still not solved.

Reply:
Is having an aarch64 build required? 

If not, then if I were to build an aarch64 binary separately and put it in the main apps release page and then used it for aarch64 flatpak builds would that solve this issue?

Reply:
It's in the documentation you acknowledged having read, and also the project you linked as examples:
https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64

Reply:
alright this should be resolved
I apologize for missing this